app/furniture/marketplace/square_order.rb

```diff
@@ -0,0 +1,149 @@
+# SquareOrder is a service class for coordinating and executing interactions
+# between a Convene Marketplace and a seller's Square account.
+#
+# Current behavior is limited to registering orders in a seller's account upon
+# sale so that we help synchronize record keeping for Marketplace owners and
+# sellers.
+#
+# NOTE: the simple `Markeplace > SquareOrder` namespacing was chosen
+# to avoid collision with the `Square` namespace already included by Square's gem.
+class Marketplace
+  class SquareOrder
+    def initialize(order)
+      @order = order
+      @marketplace = order.marketplace
+      @shopper = order.shopper
+      @space = order.marketplace.space
+      @ordered_products = @order.ordered_products
+      @square_location_id = @marketplace.settings["square_location_id"]
+      @square_order_id = nil
+      @square_payment_id = nil
+    end
+
+    def send_to_seller_dashboard
+      square_create_order_response = create_square_order
+
+      if square_create_order_response
+        @square_order_id = square_create_order_response.body.order[:id]
+        square_create_payment_response = create_square_order_payment
+        @square_payment_id = square_create_payment_response.body.payment[:id]
+
+        # This data is intended for use in debugging, etc... until we further
+        # the Square integration productize
+        {
+          order_id: @square_order_id,
+          payment_id: @square_payment_id
+        }
+      else
+        # TODO: Noop for now
+      end
+    end
+
+    # Square sets max of 43 chars
+    private def square_idemp_key
+      "#{@order.id}_#{8.times.map { rand(10) }.join}"
+    end
+
+    private def create_square_order
```

Comment:
Could we log an order event before and after attempting to send to Square so that we have a record of attempts to send the Order to Square?

Reply:
To clarify, I mean record an `order.events` object in the database; not just a Rails logger.